plugins/crd/plugin_impl_crd.go

```diff
@@ -0,0 +1,345 @@
+
+	if err != nil {
+		plugin.Log.Errorf("Error creating api client: %s", err.Error())
+	}
+
+	// We use a shared informer from the informer factory, to save calls to the
+	// API as we grow our application and so state is consistent between our
+	// control loops. We set a resync period of 30 seconds, in case any
+	// create/replace/update/delete operations are missed when watching
+	sharedFactoryNSC = factory.NewSharedInformerFactory(plugin.crdClient, time.Second*30)
+
+	informer := sharedFactoryNSC.Networkservice().V1().NetworkServiceChannels().Informer()
+	// we add a new event handler, watching for changes to API resources.
+	informer.AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: networkservicechannelEnqueue,
+			UpdateFunc: func(old, cur interface{}) {
+				if !reflect.DeepEqual(old, cur) {
+					networkservicechannelEnqueue(cur)
+				}
+			},
+			DeleteFunc: networkservicechannelEnqueue,
+		},
+	)
+
+	// Start the informer. This will cause it to begin receiving updates from
+	// the configured API server and firing event handlers in response.
+	sharedFactoryNSC.Start(stopCh)
+	plugin.Log.Info("Started NetworkServiceChannel informer factory.")
+
+	// Wait for the informer cache to finish performing it's initial sync of
+	// resources
+	if !cache.WaitForCacheSync(stopCh, informer.HasSynced) {
+		plugin.Log.Errorf("Error waiting for informer cache to sync: %s", err.Error())
+	}
+
+	plugin.Log.Info("NetworkServiceChannel Informer is ready")
+}
+
+func informerNetworkserviceendpoints(plugin *Plugin) {
+	var err error
+
+	if err != nil {
+		plugin.Log.Errorf("Error creating api client: %s", err.Error())
+	}
+
+	// We use a shared informer from the informer factory, to save calls to the
+	// API as we grow our application and so state is consistent between our
+	// control loops. We set a resync period of 30 seconds, in case any
+	// create/replace/update/delete operations are missed when watching
+	sharedFactoryNSE = factory.NewSharedInformerFactory(plugin.crdClient, time.Second*30)
+
+	informer := sharedFactoryNSE.Networkservice().V1().NetworkServiceEndpoints().Informer()
+	// we add a new event handler, watching for changes to API resources.
+	informer.AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: networkserviceendpointEnqueue,
+			UpdateFunc: func(old, cur interface{}) {
+				if !reflect.DeepEqual(old, cur) {
+					networkserviceendpointEnqueue(cur)
+				}
+			},
+			DeleteFunc: networkserviceendpointEnqueue,
+		},
+	)
+
+	// Start the informer. This will cause it to begin receiving updates from
+	// the configured API server and firing event handlers in response.
+	sharedFactoryNSE.Start(stopCh)
+	plugin.Log.Info("Started NetworkServiceEndpoints informer factory.")
+
+	// Wait for the informer cache to finish performing it's initial sync of
+	// resources
+	if !cache.WaitForCacheSync(stopCh, informer.HasSynced) {
+		plugin.Log.Errorf("Error waiting for informer cache to sync: %s", err.Error())
```

Comment:
Handle errors by creating a channel for each of these. Use a Workgroup here and have each routine call done() on these when they exit successfully.